http_request_parser.cc

```diff
@@ -0,0 +1,264 @@
+#include <sstream>
+#include "http_request_parser.h"
+#include "http_request.h"
+
+
+namespace http {
+
+// Default constructor
+request_parser::request_parser() {
+    reset();
+}
+
+
+// Performs a URL-decoding on a string and returns true if succeeded
+bool request_parser::decode_url(const std::string& in, std::string& out) {
+    out.clear();
+    out.reserve(in.size());
+    for (std::size_t i = 0; i < in.size(); ++i) {
+        if (in[i] == '%') {
+            if (i + 3 <= in.size()) {
+                int value = 0;
+                std::istringstream is(in.substr(i + 1, 2));
+                if (is >> std::hex >> value) {
+                    out += static_cast<char>(value);
+                    i += 2;
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        } else if (in[i] == '+') {
+            out += ' ';
+        } else {
+            out += in[i];
+        }
+    }
+    return true;
+}
+
+
+// Preps the parser to start on a whole new input
+void request_parser::reset() {
+    state = method_start;
+}
+
+
+// Parses data
+request_parser::result request_parser::consume(request& req,
+                                                      char input) {
```

Comment:
On a new line here,  but it should be on same line as the other function parameters.